File: opengl/src/main/java/com/example/opengl/gl/filter/IFilter.java
package com.example.opengl.gl.filter;

import android.opengl.GLSurfaceView;

/**
 * @author lmx
 * Created by lmx on 2019/5/9.
 */
public interface IFilter extends GLSurfaceView.Renderer
{
    int Flip_Horizontal = 1;//水平翻转
    int Flip_Vertical = 1 << 1;//垂直翻转

    void onRelease();
}

File: app/src/main/java/com/maoxin/apkshell/camera/filter/IFilter.kt
package com.maoxin.apkshell.camera.filter

import android.content.Context

/** @author lmx
 * Created by lmx on 2020/9/7.
 */
interface IFilter {
    fun onCreate(context: Context)
    fun onSizeChange(context: Context, cameraWidth: Int, cameraHeight: Int, viewWidth: Int, viewHeight: Int, degree: Int, flip: Int)
    fun onDestroy(context: Context)
    fun onDraw(width: Int, height: Int, commonVao: Int, vararg textures: Int):Int
}